test/e2e_source_map_test.ts

```diff
@@ -367,6 +368,587 @@ function createTests(useTransformer: boolean) {
+  //   sources.set('input.ts', `enum Direction {
+  //       Up = 1,
+  //       Down,
+  //       Left,
+  //       Right
+  //     }`);
+
+  //   // Run tsickle+TSC to convert inputs to Closure JS files.
+  //   const {compiledJS, sourceMap} = compile(sources, {useTransformer});
+
+  //   {
+  //     const {line, column} = getLineAndColumn(compiledJS, '/** @enum {number} */');
+  //     expect(sourceMap.originalPositionFor({line, column}).line)
+  //         .to.equal(1, 'enum type annotation');
+  //     expect(sourceMap.originalPositionFor({line, column}).source)
+  //         .to.equal('input.ts', 'input file name');
+  //   }
+  //   {
+  //     const {line, column} = getLineAndColumn(compiledJS, 'var Direction');
+  //     expect(sourceMap.originalPositionFor({line, column}).line).to.equal(1, 'enum declaration');
+  //     expect(sourceMap.originalPositionFor({line, column}).source)
+  //         .to.equal('input.ts', 'input file name');
+  //   }
+  //   {
+  //     const {line, column} = getLineAndColumn(compiledJS, 'Up: 1,');
+  //     expect(sourceMap.originalPositionFor({line, column}).line)
+  //         .to.equal(2, 'enum option definition');
+  //     expect(sourceMap.originalPositionFor({line, column}).source)
+  //         .to.equal('input.ts', 'input file name');
+  //   }
+  //   {
+  //     const {line, column} = getLineAndColumn(compiledJS, 'Direction[Direction.Up] = "Up";');
+  //     expect(sourceMap.originalPositionFor({line, column}).line)
+  //         .to.equal(2, 'enum option name declaration');
+  //     expect(sourceMap.originalPositionFor({line, column}).source)
+  //         .to.equal('input.ts', 'input file name');
+  //   }
+  // });
+
+  it('maps template spans', () => {
+    const sources = new Map<string, string>();
+    sources.set('input.ts', '`text text text\n${1 + 1}\n${null as string}`');
+
+    // Run tsickle+TSC to convert inputs to Closure JS files.
+    const {compiledJS, sourceMap} = compile(sources, {useTransformer});
+
+    {
+      const {line, column} = getLineAndColumn(compiledJS, 'text text text');
+      expect(sourceMap.originalPositionFor({line, column}).line).to.equal(1, 'basic text');
+      expect(sourceMap.originalPositionFor({line, column}).source)
+          .to.equal('input.ts', 'input file name');
+    }
+    {
+      const {line, column} = getLineAndColumn(compiledJS, '1 + 1');
+      expect(sourceMap.originalPositionFor({line, column}).line).to.equal(2, 'addition');
+      expect(sourceMap.originalPositionFor({line, column}).source)
+          .to.equal('input.ts', 'input file name');
+    }
+    {
+      const {line, column} = getLineAndColumn(compiledJS, 'null');
+      expect(sourceMap.originalPositionFor({line, column}).line)
+          .to.equal(3, 'type cast with parentheses');
+      expect(sourceMap.originalPositionFor({line, column}).source)
+          .to.equal('input.ts', 'input file name');
+    }
+  });
+
+  it('maps type assertions', () => {
```

Comment:
How many of these tests actually test something that follows a different code path in the underlying implementation? Tests are good, but if they don't really test anything different, we'll just slow down our test suite and will have to maintain these in the future.